crates/typst/src/doc.rs

```diff
@@ -407,6 +450,35 @@ impl Debug for Frame {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub enum FrameKind {
+    /// A container which follows its parent's size.
+    Soft,
+
+    /// A container which uses its own size.
+    Hard,
+}
+
+impl FrameKind {
+    /// Returns `true` if the frame is soft.
+    pub fn is_soft(self) -> bool {
+        matches!(self, Self::Soft)
+    }
+
+    /// Returns `true` if the frame is hard.
+    pub fn is_hard(self) -> bool {
+        matches!(self, Self::Hard)
+    }
+}
+
+impl Default for FrameKind {
```

Comment:
I kept soft as a default but with a better justification. I chose `Soft` because it's the most common frame type, but beyond that, it's also the type that does not "touch" the hierarchy for gradients. Therefore imo it's the best preset.